Blob type fails to compile with mongo

* **Overview of the issue**

Blob type fails to compile with mongo
```
[INFO] --- maven-processor-plugin:2.2.4:process (process) @ sample-mongo ---

[ERROR] diagnostic: /home/travis/app-mongodb/src/main/java/com/mycompany/myapp/domain/BankAccount.java:62: error: cannot find symbol

    @Column(name = "attachment_content_type", nullable = false)

     ^

  symbol:   class Column

  location: class com.mycompany.myapp.domain.BankAccount
```

* **JHipster Version(s)** 
Any version including latest master

* **JHipster configuration, a `.yo-rc.json` file generated in the root folder** 
https://github.com/jhipster/generator-jhipster/blob/master/travis/samples/app-mongodb/.yo-rc.json

* **Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory** 
```json
{
    "fields": [
        {
            "fieldId": 1,
            "fieldName": "attachment",
            "fieldType": "byte[]",
            "fieldTypeBlobContent": "any"
        }
    ],
    "changelogDate": "20150805124838",
    "dto": "yes",
    "pagination": "no"
}
```

* **Reproduce the error** 
```
mvn test
```

* **Suggest a Fix**

The annotation should probably be ```@Field```, not ```@Column```

--- a/entity/templates/src/main/java/package/domain/_Entity.java
+++ b/entity/templates/src/main/java/package/domain/_Entity.java
@@ -97,9 +97,10 @@ public class <%= entityClass %> implements Serializable {
     <%_ } _%>
     private <%= fields[fieldId].fieldType %> <%= fields[fieldId].fieldName %>;
 
-    <%_ if (fields[fieldId].fieldType == 'byte[]') { _%>
-
-    @Column(name = "<%=fields[fieldId].fieldNameUnderscored %>_content_type"<% if (required) { %>, nullable = false<% } %>)
+    <%_ if (fields[fieldId].fieldType == 'byte[]') { _%><%_ if (databaseType == 'sql') { _%>
+    @Column(name = "<%=fields[fieldId].fieldNameUnderscored %>_content_type"<% if (required) { %>, nullable = false<% } %>) <%_ } _%>
+    <% if (databaseType == 'mongodb') { %>@Field("<%=fields[fieldId].fieldNameUnderscored %>_content_type")
+    <%_ } _%>
     private String <%= fields[fieldId].fieldName %>ContentType;
     <%_ }
     }
@@ -153,6 +154,7 @@ public class <%= entityClass %> implements Serializable {
     private <%= relationships[relationshipId].otherEntityNameCapitalized %> <%= relationships[relationshipId].relationshipFieldName %>;
 
     <%_ } } _%>
+    
     public <% if (databaseType == 'sql') { %>Long<% } %><% if (databaseType == 'mongodb') { %>String<% } %><% if (databaseType == 'cassandra') { %>UUID<% } %> getId() {
         return id;
     }

--- a/travis/samples/app-mongodb/.jhipster/BankAccount.json
+++ b/travis/samples/app-mongodb/.jhipster/BankAccount.json
@@ -56,6 +56,12 @@
             "fieldName": "accountType",
             "fieldType": "BankAccountType",
             "fieldValues": "CHECKING,SAVINGS,LOAN"
+        },
+        {
+            "fieldId": 11,
+            "fieldName": "attachment",
+-            "fieldType": "byte[]",
+-            "fieldTypeBlobContent": "any"
         }
     ],
     "changelogDate": "20150805124838",

--- a/travis/samples/app-mongodb/.jhipster/BankAccount.json
+++ b/travis/samples/app-mongodb/.jhipster/BankAccount.json
@@ -60,8 +60,8 @@
         {
             "fieldId": 11,
             "fieldName": "attachment",
--            "fieldType": "byte[]",
--            "fieldTypeBlobContent": "any"
+            "fieldType": "byte[]",
+            "fieldTypeBlobContent": "any"
         }
     ],
     "changelogDate": "20150805124838",

